test/integration/prerender-preview/test/index.test.js

```diff
@@ -0,0 +1,186 @@
+    expect(res.status).toBe(200)
+
+    const cookies = res.headers
+      .get('set-cookie')
+      .split(',')
+      .map(cookie.parse)
+
+    expect(cookies.length).toBe(2)
+    expect(cookies[0]).toMatchObject({ Path: '/', SameSite: 'Strict' })
+    expect(cookies[0]).toHaveProperty('__prerender_bypass')
+    expect(cookies[0]).not.toHaveProperty('Max-Age')
+    expect(cookies[1]).toMatchObject({ Path: '/', SameSite: 'Strict' })
+    expect(cookies[1]).toHaveProperty('__next_preview_data')
+    expect(cookies[1]).not.toHaveProperty('Max-Age')
+
+    previewCookieString =
+      cookie.serialize('__prerender_bypass', cookies[0].__prerender_bypass) +
+      '; ' +
+      cookie.serialize('__next_preview_data', cookies[1].__next_preview_data)
+  })
+
+  it('should return fallback page on preview request', async () => {
+    const res = await fetchViaHTTP(
+      appPort,
+      '/',
+      {},
+      { headers: { Cookie: previewCookieString } }
+    )
+    const html = await res.text()
+
+    const { nextData, pre } = getData(html)
+    expect(nextData).toMatchObject({ isFallback: true })
+    expect(pre).toBe('Has No Props')
+  })
+
+  it('should return cookies to be expired on reset request', async () => {
+    const res = await fetchViaHTTP(
+      appPort,
+      '/api/reset',
+      {},
+      { headers: { Cookie: previewCookieString } }
+    )
+    expect(res.status).toBe(200)
+
+    const cookies = res.headers
+      .get('set-cookie')
+      .split(',')
+      .map(cookie.parse)
+
+    expect(cookies.length).toBe(2)
+    expect(cookies[0]).toMatchObject({
+      Path: '/',
+      SameSite: 'Strict',
+      'Max-Age': '0',
+    })
+    expect(cookies[0]).toHaveProperty('__prerender_bypass')
+    expect(cookies[1]).toMatchObject({
+      Path: '/',
+      SameSite: 'Strict',
+      'Max-Age': '0',
+    })
+    expect(cookies[1]).toHaveProperty('__next_preview_data')
+  })
+
+  /** @type import('next-webdriver').Chain */
+  let browser
+  it('should start the client-side browser', async () => {
+    browser = await webdriver(
+      appPort,
+      '/api/preview?' + qs.stringify({ client: 'mode' })
+    )
+  })
+
+  it('should fetch preview data', async () => {
+    await browser.get(`http://localhost:${appPort}/`)
+    await browser.waitForElementByCss('#props-pre')
+    expect(await browser.elementById('props-pre').text()).toBe('Has No Props')
+    await new Promise(resolve => setTimeout(resolve, 2000))
+    expect(await browser.elementById('props-pre').text()).toBe(
+      'true and {"client":"mode"}'
+    )
+  })
+
+  it('should fetch prerendered data', async () => {
+    await browser.get(`http://localhost:${appPort}/api/reset`)
+
+    await browser.get(`http://localhost:${appPort}/`)
+    await browser.waitForElementByCss('#props-pre')
+    expect(await browser.elementById('props-pre').text()).toBe(
+      'undefined and undefined'
+    )
+  })
+
+  it('should close the client-side browser', async () => {
+    await browser.close()
```

Comment:
```suggestion
```
This doesn't actually do anything anymore and it is closed by the jest-environment now since we re-use the browser instance

Reply:
It feels wrong to leave the reference hanging. I'd rather keep the `await browser.close()`, even if it's noop. Imagine if we change it in the future!